roles/horizon/tasks/main.yml

```diff
@@ -1,121 +1,92 @@
+- name: create dashboard virtualhost on other
+  template: |
+    src=etc/apache2/sites-available/openstack_dashboard.conf
+    dest=/etc/apache2/sites-available/openstack_dashboard.conf
   when: ansible_distribution_version != "12.04"
   notify:
-    - restart apache
+    - reload apache
 
 - name: enable horizon apache site
-  command: a2ensite openstack_dashboard
+  apache2_site: state=enabled name=openstack_dashboard
   notify:
-    - restart apache
-
-- name: static asset directories
-  file: dest={{ item }}
-        state=directory
-        owner=www-data
-        group=www-data
-        mode=0755
+    - reload apache
+
+- name: create static asset dirs
+  file: |
+    dest={{ item }}
+    state=directory
+    owner=www-data
+    group=www-data
+    mode=0755
   with_items:
     - /opt/stack/horizon/static
     - /opt/stack/horizon/static/dashboard
 
-- name: horizon local settings
-  template: src=opt/stack/horizon/openstack_dashboard/local/local_settings.py
-            dest=/opt/stack/horizon/openstack_dashboard/local/local_settings.py
-            mode=0644
+- name: dashboard settings
+  template: |
+    src=opt/stack/horizon/openstack_dashboard/local/local_settings.py
+    dest=/opt/stack/horizon/openstack_dashboard/local/local_settings.py
+    mode=0644
   notify:
-    - restart apache
+    - reload apache
 
 - name: custom horizon logo
-  get_url: url={{ horizon.logo_url }}
-           dest=/opt/stack/horizon/openstack_dashboard/static/dashboard/img/logo.png
-           mode=0644 force=yes
+  get_url: url={{ horizon.logo_url }} dest=/opt/stack/horizon/openstack_dashboard/static/dashboard/img/logo.png mode=0644 force=yes
 
 - name: custom horizon splash logo
-  get_url: url={{ horizon.logo_url }}
-           dest=/opt/stack/horizon/openstack_dashboard/static/dashboard/img/logo-splash.png
-           mode=0644 force=yes
+  get_url: url={{ horizon.logo_url }} dest=/opt/stack/horizon/openstack_dashboard/static/dashboard/img/logo-splash.png mode=0644 force=yes
 
 - name: custom horizon favicon
-  get_url: url={{ horizon.favicon_url }}
-           dest=/opt/stack/horizon/openstack_dashboard/static/dashboard/img/favicon.ico
-           force=yes
+  get_url: url={{ horizon.favicon_url }} dest=/opt/stack/horizon/openstack_dashboard/static/dashboard/img/favicon.ico force=yes
 
 - name: put images and fonts where apache can find them
-  file: src=/opt/stack/horizon/openstack_dashboard/static/dashboard/{{ item }}
-        dest=/opt/stack/horizon/static/dashboard/{{ item }}
-        state=link
-        owner=www-data
-        group=www-data
+  file: |
+    src=/opt/stack/horizon/openstack_dashboard/static/dashboard/{{ item }}
+    dest=/opt/stack/horizon/static/dashboard/{{ item }}
+    state=link
+    owner=www-data
+    group=www-data
   with_items:
     - img
     - fonts
 
-# flush before ensuring apache running, saves immediate restart
-- meta: flush_handlers
-
-- name: ensure apache started
+- name: ensure apache is running
```

Comment:
Now you have a start followed by an immediate reload. Will apache be okay with that?